server/data/data-jmap/src/test/java/org/apache/james/jmap/api/preview/PreviewTest.java

```diff
@@ -70,4 +74,73 @@ void fromShouldNotThrowWhenValueLengthIsEqualsToMaximum256() {
         assertThatCode(() -> Preview.from(Strings.repeat("a", 256)))
             .doesNotThrowAnyException();
     }
+
+    @Nested
+    class ComputeTest {
+
+        @Test
+        void computeShouldReturnStringEmptyWhenEmptyTextBody() throws Exception {
+            assertThat(Preview.compute(Optional.empty()))
+                .isEqualTo(Preview.NO_BODY);
+        }
+    
+        @Test
+        void computeShouldReturnStringEmptyWhenStringEmptyTextBody() throws Exception {
+            assertThat(Preview.compute(Optional.of("")))
+                .isEqualTo(Preview.NO_BODY);
+        }
+    
+        @Test
+        void computeShouldReturnStringEmptyWhenOnlySpaceTabAndBreakLines() throws Exception {
+            assertThat(Preview.compute(Optional.of(" \n\t ")))
+                .isEqualTo(Preview.NO_BODY);
+        }
+    
+        @Test
+        void computeShouldReturnStringEmptyWhenOnlySpace() throws Exception {
+            assertThat(Preview.compute(Optional.of(" ")))
+                .isEqualTo(Preview.NO_BODY);
+        }
+    
+        @Test
+        void computeShouldReturnStringEmptyWhenOnlyTab() throws Exception {
+            assertThat(Preview.compute(Optional.of("\t")))
+                .isEqualTo(Preview.NO_BODY);
+        }
+    
+        @Test
+        void computeShouldReturnStringEmptyWhenOnlyBreakLines() throws Exception {
+            assertThat(Preview.compute(Optional.of("\n")))
+                .isEqualTo(Preview.NO_BODY);
+        }
+    
+        @Test
+        void computeShouldReturnStringWithoutTruncation() throws Exception {
+            String body = StringUtils.leftPad("a", 100, "b");
+    
+            assertThat(Preview.compute(Optional.of(body))
+                    .getValue())
+                .hasSize(100)
+                .isEqualTo(body);
+        }
+    
+        @Test
+        void computeShouldReturnStringIsLimitedTo256Length() throws Exception {
+            String body = StringUtils.leftPad("a", 300, "b");
+            String expected = StringUtils.leftPad("b", Preview.MAX_LENGTH, "b");
```

Comment:
please do not reference the constant in Preview, we should prevent accidental refactoring from break both the code and the test at the same time.